packages/nimble-components/src/patterns/button/styles.ts

```diff
@@ -257,84 +265,160 @@ export const buttonAppearanceVariantStyles = css``.withBehaviors(
+            }
+
+            :host([appearance-variant='primary'][disabled])
+                slot[name='end']::slotted(*) {
+                ${iconColor.cssCustomProperty}: ${buttonLabelFontColor};
+            }
+
+            :host([appearance-variant='accent']) .control {
+                background-color: ${buttonAccentBackgroundColor};
+                color: ${buttonBlockAccentFontColor};
+            }
+
+            :host([appearance-variant='accent']) .control:hover {
+                background-color: transparent;
+                background-image: linear-gradient(
+                    ${buttonAccentBackgroundColor},
+                    ${buttonAccentBackgroundColor}
+                );
+                background-size: calc(100% - 4px) calc(100% - 4px);
+                color: ${buttonBlockAccentFontColor};
+            }
+
+            :host([appearance-variant='accent']) .control${focusVisible} {
+                background-color: transparent;
+                background-image: linear-gradient(
+                    ${buttonAccentBackgroundColor},
+                    ${buttonAccentBackgroundColor}
+                );
+                background-size: calc(100% - 4px) calc(100% - 4px);
+            }
+
+            :host([appearance-variant='accent']) .control:active {
+                background-image: linear-gradient(
+                    ${buttonAccentFillActionColor},
+                    ${buttonAccentFillActionColor}
+                );
+                background-size: calc(100% - 2px) calc(100% - 2px);
+            }
+
+            :host([appearance-variant='accent'][disabled]) .control {
+                background-color: rgba(${borderRgbPartialColor}, 0.1);
                 border-color: transparent;
-                border-width: ${borderWidth};
-                box-shadow: none;
-                padding: 0 ${standardPadding};
+                color: rgba(${actionRgbPartialColor}, 0.3);
+                background-image: none;
+            }
+
+            :host([appearance-variant='accent']) [part='start'] {
+                ${iconColor.cssCustomProperty}: white;
+            }
+
+            :host([appearance-variant='accent'][disabled])
+                slot[name='start']::slotted(*) {
+                ${iconColor.cssCustomProperty}: ${buttonLabelFontColor};
             }
 
-            :host([appearance-variant='primary'][disabled][content-hidden])
-                .control {
-                padding: 0px;
+            :host([appearance-variant='accent']) [part='end'] {
+                ${iconColor.cssCustomProperty}: white;
             }
 
-            :host([appearance-variant='primary']) .control:active::before {
-                border-color: ${fillSelectedColor};
-                border-width: ${borderWidth};
+            :host([appearance-variant='accent'][disabled])
+                slot[name='end']::slotted(*) {
+                ${iconColor.cssCustomProperty}: ${buttonLabelFontColor}
             }
```

Comment:
make sure your styles are in dom order make sure your styles are in dom order: https://github.com/ni/nimble/blob/main/packages/nimble-components/docs/css-guidelines.md#group-selectors-by-target-and-document-order

The grouping by appearance variant is fine (technically not aligned with the guidelines) but within each of the appearance varaint groups make sure they are in dom order